forge-gui/res/cardsfolder/a/accomplished_automaton.txt

```diff
@@ -3,5 +3,5 @@ ManaCost:7
 Types:Artifact Creature Construct
 PT:5/7
 K:Fabricate:1
-DeckHas:Ability$Counters|Token
+DeckHas:Ability$Counters|Token & Type$Servo
```

Comment:
I wanna say for consistency this should rather be done via engine support as well
At least it seems not too difficult, just not sure it's in my schedule for the rest of the year :p
So probably better to skip the double work (sorry)

Reply:
The same would be valid for the likes of embalm and eternalize, then?

Reply:
well those aren't just tokenscripts keyworded which my approach could handle
while still tokens I'd need to support loose params in general, I'm not even sure that would be worth it yet

Reply:
Oh my mistake then: it's living weapon and for Mirrodin that are relevant here. Will clean up.

Reply:
And just so it's clear: is `Investigate` something that your approach can handle? I ask because I notice that, in the code, it strays a bit from the pattern for fabricate, living weapon and the like.

Reply:
not sure what you mean :)

Reply:
Well, looking over [InvestigateEffect.java](https://github.com/Card-Forge/forge/blob/master/forge-game/src/main/java/forge/game/ability/effects/InvestigateEffect.java), `c_a_clue_draw` shows up but it isn't associated with the `StringBuilder` element in a comparable way to the keyword abilities we've discussed (as seen on [CardFactoryUtil.java](https://github.com/Card-Forge/forge/blob/master/forge-game/src/main/java/forge/game/card/CardFactoryUtil.java)). At least not so clearly associated that I feel confident reverting the relevant work I've done. So I'd like confirmation that, for instance, the` DeckHas:` line on [Alquist Proft, Master Sleuth](https://github.com/dracontes/forge/blob/rb-editpile-decktag-a-1-patch-2/forge-gui/res/cardsfolder/a/alquist_proft_master_sleuth.txt) is either to remove or to keep.

Reply:
Right, that would actually be a third variant to support 😅

I might consider a more generic approach for this one where we can hardcode some implication logic for Ability$ values, e.g. `Investitigate -> Color$Colorless & Type$Artifact|Clue` and potentially store them in an extra file

but that's also something I'd only attempt after implementing what I originally commented on first 🤹‍♂️